hack/docs/supported-versions.sh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+
+set -eu
+
+# Loosely based on https://github.com/argoproj/argo-cd/blob/5b79c34c72300e6e2e6336051ce6992f6d54011c/hack/update-supported-versions.sh
+
+# Extract major/minor versions from branch name, e.g. "release-3.5" will become:
+# release-
+# 3
+# 5
+mapfile -t branch_parts < <(git rev-parse --abbrev-ref ${1:-HEAD} | grep -Eo '.*release-|[0-9]')
+
+if [[ -z "${branch_parts[@]}" ]]; then
+  echo 'This page is populated for released Argo Workflows versions. Use the version selector to view this table for a specific version.'
+  exit
+fi
+
+echo "The following table shows the versions of Kubernetes that are tested with each version of Argo Workflows."
+echo
+echo "| Argo Workflows version | Kubernetes versions |"
+echo "|------------------------|---------------------|"
+
+for n in 0 1 2; do
```

Comment:
Why iterate back like this at all? Why not force the user to look at the page for the version they are using instead?

The table for other release numbers may be wrong as an update to k8s version may have occurred on their branch but a release not happened. If we cherry-pick this back to 3.5 then it will iterate back to 3.3 which is even more unsupported than 3.4, and at some point this breaks with a 4.0 release. And at 3.7 release 3.6 will continue to want to document 3.4.

I'd rather we always displayed the "version selector" sentence, and just showed the current branch.

We do lose the version information for `main`/`latest` which is probably a healthy thing, so I agree with that.

Reply:
If a user is on an older Kubernetes version and they want to determine what version(s) of Argo Workflows they can use, then a table provides a better user experience than manually switching between versions. But that's probably not common enough to be worth the extra complexity, so I went ahead and rewrote this to only print a single sentence with the tested Kubernetes versions for the current version. That means `hack/k8s-versions.sh` no longer needs to be compatible with old release branches, so I updated it to be a simple config file that defines the minimum and maximum Kubernetes versions, which is cleaner.